### DISCUSSION TITLE
Call JavaScript new operator from Python?

### DISCUSSION THREAD

**Discussion by johnwason** [pyodide/pyodide#1273] [Q&A]
Is it possible to call the JavaScript `new` operator from Python? I see how to call the function by importing from the `js` module, but not how to construct a new JavaScript object.

**Comment by rth** [comment]
Yes it's possible. `new X(..)` in JS is equivalent to `X.new(..)` from Python. See the [documentation](https://pyodide.readthedocs.io/en/latest/usage/type-conversions.html#javascript-from-python) for more details.

**Reply by johnwason** [reply]
I am getting this error:

`AttributeError: 'JsBoundMethod' object has no attribute 'new'`

I am trying to construct a 3rd party JS library object.

**Reply by hoodmane** [reply]
This is known bug in the released versions. It is fixed in the development version of pyodide. The problem is that a normal `JsProxy` has a `new` method, but if you get the function by indexing a `JsProxy` it gives a `JsBoundMethod` which does not have a `new` method. A workaround would be:

```js
window.SomeType = somepackage.SomeType;
pyodide.runPython(`
from js import SomeType
SomeType.new("blah")
`);
```
